lib/screens/v2/widget_instance/subway_status.ex

```diff
@@ -468,8 +455,44 @@ defmodule Screens.V2.WidgetInstance.SubwayStatus do
       |> Enum.uniq()
       |> Enum.sort()
 
-    MapSet.size(alert_trunk_stops) > 0 or
-      alert_whole_line_stops == @green_line_branches
+    alert_whole_line_stops == @green_line_branches
+  end
+
+  # If any closed stop is served by more than one branch, the alert affects the trunk
+  defp alert_affects_gl_trunk?(
+         %Alert{
+           effect: :station_closure,
+           informed_entities: informed_entities
+         },
+         gl_stop_sequences
+       ) do
+    informed_entities
+    |> Enum.map(fn e -> Map.get(e, :stop) end)
+    |> Enum.reject(&is_nil/1)
+    |> Enum.uniq()
+    |> Enum.any?(fn informed_stop ->
+      Enum.count(gl_stop_sequences, &(informed_stop in &1)) > 1
+    end)
+  end
+
+  #
+  defp alert_affects_gl_trunk?(%Alert{informed_entities: informed_entities}, gl_stop_sequences) do
+    alert_stops =
+      informed_entities
+      |> Enum.filter(fn
+        %{stop: nil} -> false
+        ie -> String.starts_with?(ie.stop, "place-") and String.starts_with?(ie.route, "Green-")
+      end)
+      |> Enum.uniq()
+
```

Comment:
Is `alert_stops` intended to be a list of informed entities? It looks like it should be a list of stop IDs since its elements get checked against `gl_stop_sequences`.

Reply:
Yikes. Good catch. Not sure how I missed that.